solidity/random-beacon/contracts/libraries/DKG.sol

```diff
@@ -122,14 +125,15 @@ library DKG {
         uint8[] misbehavedMembersIndices,
         bytes signatures,
         uint256[] signingMembersIndices,
-        uint32[] members
+        bytes32 membersHash
```

Comment:
The last time I checked, emitting `uint32[] members` in an event had a significant gas cost, it does not come for free and it is possible some savings we got from the change in this PR come from this change.

As much as I'd like to make it cheaper, I do not know if there is an alternative if we do not want to execute view functions on a previous chain state to know who is in the group.

Reply:
Yes, the fewer events we have, the cheaper it is. I will try to track the exact cost of emitting such an event but need to fix the tests first.

Reply:
It's crucial for the challenge submission to know exactly what was submitted as DKG Result. A DKG result submitter can provide any set of members, which would be needed for a challenge. Knowing just the hash wouldn't be enough, the challenger should pull the members' list from the DKG result submission transaction input parameters, which adds complexity. We should first ensure that it will be possible to obtain all the required information from the transaction, if they are not emitted in the event.

Reply:
The emitted event should have a transaction hash, we could get the transaction by the hash and try decoding input parameters. It's more complicated than emitting data in an event, but possible. This way we could reduce costs even more by cutting out other parameters from events, that duplicate transaction's input paramters.

Reply:
That's true, we need group members during the challenge step [here](https://github.com/keep-network/keep-core/blob/group-members-storage-optimization/solidity/random-beacon/contracts/DKGValidator.sol#L188). We do emit all the members in `DkgResultApproved` event, maybe we can emit members (excluding IA/DQ) in `DkgResultSubmitted`, but emit only `hash(members)` in the `DkgResultApproved` event. I'll see what's the cost of that.

Reply:
gas used for `DkgResultSubmitted` with `hash(result.members)` for group size 64: 44980

gas used for `DkgResultSubmitted` with `result.members` for group size 64: 61065

The diff is 16k and having in mind that it would be easier to read the data from event rather than the input params from the DKG result submission, I think we should leave `uint32[] members`.

Reply:
Here's the change 2b583e9 What I'm not 100% sure about is if we should emit members hash with or without misbehaved members in `DkgResultApproved` event. Thoughts? For submitters or callers that would pass members hash in the functions param, it would be easier, but on our side, we'll add gas to filter out misbehaved members. On the other hand, if the second PR lands in main which deals with already filtered out members, then we won't be needing filtering out members here.